aiogram_dialog/widgets/kbd/datecalendar.py

```diff
@@ -0,0 +1,151 @@
+                              manager: DialogManager) -> List[List[InlineKeyboardButton]]:
+        offset = self.get_offset(manager)
+        if manager.context.data(f"{self.widget_id}:LEVEL", None, internal=True) == YEARS_LEVEL:
+            rendered_kbd = self.years_kbd(offset)
+        elif manager.context.data(f"{self.widget_id}:LEVEL", None, internal=True) == MONTHS_LEVEL:
+            rendered_kbd = self.months_kbd(offset) 
+        else:
+            rendered_kbd = self.days_kbd(offset)
+
+        return rendered_kbd
+
+    async def process_callback(self, 
+                               c: CallbackQuery, 
+                               dialog: Dialog,
+                               manager: DialogManager) -> bool:
+        prefix = f"{self.widget_id}:"
+        offset = self.get_offset(manager)
+        if not c.data.startswith(prefix):
+            return False
+        data = c.data[len(prefix):]
+        if data == "+":
+            self.set_offset(datetime(offset.year + (offset.month // 12), ((offset.month % 12) + 1), 1), manager)
+        elif data == "-":
+            if offset.month == 1:
+                self.set_offset(datetime(offset.year - 1, 12, 1), manager)
+            else:
+                self.set_offset(datetime(offset.year, (offset.month - 1), 1), manager)
+        elif data == MONTHS_LEVEL:
+            manager.context.set_data(f"{self.widget_id}:LEVEL", data, internal=True)
+        elif data == YEARS_LEVEL:
+            manager.context.set_data(f"{self.widget_id}:LEVEL", data, internal=True)
+        elif "month" in data:
+            data = int(c.data[len(prefix) + 6:])
+            manager.context.set_data(f"{self.widget_id}:LEVEL", False, internal=True)
+            self.set_offset(datetime(offset.year, data, 1), manager)
+        elif "year" in data:
+            data = int(c.data[len(prefix) + 5:])
+            manager.context.set_data(f"{self.widget_id}:LEVEL", MONTHS_LEVEL, internal=True)
+            self.set_offset(datetime(data, 1, 1), manager)
+        else:
+            raw_date = int(data)
+            manager.context.set_data(f"{self.widget_id}", raw_date)#selected
+            await self.on_click.process_event(c, self, manager, date.fromtimestamp(raw_date))
+
+    def years_kbd(self, offset) -> List[List[InlineKeyboardButton]]:
+        years = []
+        for n in [[*range(i, i + 3)] for i in range(offset.year - 7, offset.year + 7, 3)]:
+            year_row = []
+            for y in n:
+                year_text = date(y, 1, 1).strftime("%Y")
+                year_row.append(InlineKeyboardButton(text=f"{year_text}",
+                                                     callback_data=f"{self.widget_id}:year:{y}"))
+            years.append(year_row)
+
+        return [*years]
+
+    def months_kbd(self, offset) -> List[List[InlineKeyboardButton]]:
+        header_year = offset.strftime("year %Y")
+        months = []
+        for n in [[*range(i, i + 3)] for i in range(1, 13, 3)]:
```

Comment:
what is happening here? It is constant list of months? Can it be simplified or expanded?